### PR TITLE
feat: Implement playing

### PR DESCRIPTION
**Pull request by meczajkowski**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The task goal was to implement a feature that allows the user to play with the Tamagotchi. 

### Requirements
1. Playing should last until another action is pressed or this one is finished by clicking on the action button again.
2. Playing should recover 2 `fun` points per second.
3. Playing should decrease energy by 1 `energy` point per second.

### Code Changes Review

The code changes seem to cover all the task requirements:

1. The code checks if the Tamagotchi is currently playing and stops it if another action is pressed or if the play action is finished by clicking on the action button again.

```diff
+    // this is the code that stops the Tamagotchi from playing when another action is pressed or when the play action is finished by clicking on the action button again
     if (event.target.closest('.action-button--fun')) {
       if (this.isPlaying) {
         this.stopPlaying();
```

2. The code recovers 2 `fun` points per second while the Tamagotchi is playing.

```diff
+    // Fun
     // fun should decrease by 1 points per second
-    if (this.fun.value > 0) {
+    if (this.fun.value > 0 && !this.isPlaying) {
       this.fun.value -= 1;
+    } // Playing should recover 2 fun points per second
+    else if (this.isPlaying && this.fun.value < 9) {
+      this.fun.value += 2;
```

3. The code decreases energy by 1 `energy` point per second while the Tamagotchi is playing.

```diff
+      // Playing should decrease energy by 1 energy point per second
+      this.energy.value -= 1;
```

In conclusion, the code changes cover all the task requirements and there are no apparent bugs related to these changes.
